cloud/aws.go

```diff
@@ -35,6 +35,41 @@ func DescribeAutoScalingGroupsByNames(svc autoscalingiface.AutoScalingAPI, names
 	return result.AutoScalingGroups, nil
 }
 
+func filterAutoScalingGroupsByTag(autoScalingGroups []*autoscaling.Group, filter func([]*autoscaling.TagDescription) bool) (ret []*autoscaling.Group) {
+	for _, autoScalingGroup := range autoScalingGroups {
+		if filter(autoScalingGroup.Tags) {
+			ret = append(ret, autoScalingGroup)
+			break
+		}
+	}
+	return
+}
+
+// DescribeEnabledAutoScalingGroupsByClusterName Gets cluster AutoScalingGroups that are enabled
+// See: https://docs.aws.amazon.com/eks/latest/userguide/cluster-autoscaler.html
+func DescribeEnabledAutoScalingGroupsByClusterName(svc autoscalingiface.AutoScalingAPI, clusterName string) ([]*autoscaling.Group, error) {
+	input := &autoscaling.DescribeAutoScalingGroupsInput{}
+	var result []*autoscaling.Group
+	err := svc.DescribeAutoScalingGroupsPages(input, func(page *autoscaling.DescribeAutoScalingGroupsOutput, lastPage bool) bool {
+		tagFilter := func(tagDescriptions []*autoscaling.TagDescription) bool {
+			clusterNameTag := false
+			enabledTag := false
+			for _, tagDescription := range tagDescriptions {
+				clusterNameTag = clusterNameTag || (*tagDescription.Key == fmt.Sprintf("k8s.io/cluster-autoscaler/%s", clusterName) && *tagDescription.Value == "owned")
+				enabledTag = enabledTag || (*tagDescription.Key == "k8s.io/cluster-autoscaler/enabled" && strings.ToLower(*tagDescription.Value) == "true")
```

Comment:
There should be a function in the `aws` package imported in this file that should let you convert string pointers to a string using `aws.String(tagDescription.Key)`.

It's pretty much the equivalent of what you're doing, but in the unlikely event that for some magical reason, the key is nil, dereferencing the string pointer would cause the application to panic.

Reply:
I have fixed this.